exporter/azuremonitorexporter/trace_to_envelope.go

```diff
@@ -0,0 +1,660 @@
+}
+
+// Copies all attributes to either properties or measurements and passes the key/value to another mapping function
+func copyAndMapAttributes(
+	attributeMap pdata.AttributeMap,
+	properties map[string]string,
+	measurements map[string]float64,
+	mappingFunc func(k string, v pdata.AttributeValue)) {
+
+	attributeMap.ForEach(
+		func(k string, v pdata.AttributeValue) {
+			setAttributeValueAsPropertyOrMeasurement(k, v, properties, measurements)
+			mappingFunc(k, v)
+		})
+}
+
+// Attribute extraction logic for HTTP Span attributes
+func copyAndExtractHTTPAttributes(
+	attributeMap pdata.AttributeMap,
+	properties map[string]string,
+	measurements map[string]float64) *HTTPAttributes {
+
+	attrs := &HTTPAttributes{}
+	copyAndMapAttributes(
+		attributeMap,
+		properties,
+		measurements,
+		func(k string, v pdata.AttributeValue) { attrs.MapAttribute(k, v) })
+
+	return attrs
+}
+
+// Attribute extraction logic for RPC Span attributes
+func copyAndExtractRPCAttributes(
+	attributeMap pdata.AttributeMap,
+	properties map[string]string,
+	measurements map[string]float64) *RPCAttributes {
+
+	attrs := &RPCAttributes{}
+	copyAndMapAttributes(
+		attributeMap,
+		properties,
+		measurements,
+		func(k string, v pdata.AttributeValue) { attrs.MapAttribute(k, v) })
+
+	return attrs
+}
+
+// Attribute extraction logic for Database Span attributes
+func copyAndExtractDatabaseAttributes(
+	attributeMap pdata.AttributeMap,
+	properties map[string]string,
+	measurements map[string]float64) *DatabaseAttributes {
+
+	attrs := &DatabaseAttributes{}
+	copyAndMapAttributes(
+		attributeMap,
+		properties,
+		measurements,
+		func(k string, v pdata.AttributeValue) { attrs.MapAttribute(k, v) })
+
+	return attrs
+}
+
+// Attribute extraction logic for Messaging Span attributes
+func copyAndExtractMessagingAttributes(
+	attributeMap pdata.AttributeMap,
+	properties map[string]string,
+	measurements map[string]float64) *MessagingAttributes {
+
+	attrs := &MessagingAttributes{}
+	copyAndMapAttributes(
+		attributeMap,
+		properties,
+		measurements,
+		func(k string, v pdata.AttributeValue) { attrs.MapAttribute(k, v) })
+
+	return attrs
+}
+
+func idToHex(source []byte) string {
+	if source == nil {
+		return ""
+	}
+
+	return fmt.Sprintf("%02x", source)
+}
+
+func formatSpanDuration(span pdata.Span) string {
+	startTime := toTime(span.StartTime())
+	endTime := toTime(span.EndTime())
+	return formatDuration(endTime.Sub(startTime))
+}
+
+// Maps incoming Span to a type defined in the specification
+func mapIncomingSpanToType(attributeMap pdata.AttributeMap) spanType {
+	// No attributes
+	if attributeMap.Len() == 0 {
+		return unknownSpanType
+	}
+
+	// HTTP
+	if _, exists := attributeMap.Get(conventions.AttributeHTTPMethod); exists {
+		return httpSpanType
+	}
+
+	// RPC
+	if _, exists := attributeMap.Get(conventions.AttributeRPCSystem); exists {
+		return rpcSpanType
+	}
+
+	// Database
+	if _, exists := attributeMap.Get(conventions.AttributeDBType); exists {
```

Comment:
in the latest spec, `db.type` is renamed to `db.system` - https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/database.md#connection-level-attributes

Reply:
yeah looks like the spec was revised 2 days ago. I'll make this change.